jcontroller.php

```diff
@@ -173,6 +178,34 @@ protected function refRequest($param)
         return $this->refRequest[$param];
     }
 
+    /**
+     * Filters access to an action based on the controller methods.
+     *  
+     * If access is granted, it proceeds to display the action.
+     * If access is not granted, it displays an error code (401 - Access Forbidden).
+     * 
+     * @return void
+     */
+    public function filterAccessControl()
+    {
+        if (!isset($_GET['action'])) {
+            return;
+        }
+
+        $actionName = $_GET['action'];
+        $accessRules = $this->accessRules();
+
+        if (!array_key_exists($actionName, $accessRules)) {
+            return;
+        }
+
+        $accessMethod = $accessRules[$actionName];
+
+        if ($accessMethod != '' && !$this->$accessMethod) {
+            throw new Exception('(401 - Access Forbidden)');
```

Comment:
Please add an optional message param in your `accessRule` array. If that one is set you return the message or else use the default. Array format can be similar to this:

https://forum.yiiframework.com/t/access-rules-expression/54657